python/ray/data/preprocessors/batch_mapper.py

```diff
@@ -15,8 +16,13 @@
 if TYPE_CHECKING:
     import pandas
 
+BATCH_MAPPER_DEPRECATION_MESSAGE = (
+    "The BatchMapper preprocessor is deprecated as of Ray 2.7. "
```

Comment:
Should we raise an error? Or warn now and change this to 2.8?

Reply:
We already warned in 2.6, so this PR upgrades to an error.

Reply:
Err, should this `raise DeprecationWarning` instead then?

Reply:
Oh sorry-- I thought I misread where you left this comment.

For `BatchMapper`, this is also a new deprecation, so raising a warning here.